src/js/gi/components/PreviewBanner.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+class PreviewBanner extends React.Component {
+  render() {
+    if (!this.props.show) {
+      return null;
+    }
+
+    const headerDisplacementCSS = `
```

Comment:
Possible, but I wanted to make it explicit that I'm affecting rendering outside the visible boundaries of the component without actually touching DOM nodes outside the component.